tests/ci/create-commit.sh

```diff
@@ -0,0 +1,380 @@
+#!/bin/bash
+set -euo pipefail
+
+source /tmp/.env
+
+ID=${ID:-}
+ARCH=${ARCH:-}
+IMAGE_TYPE=${IMAGE_TYPE:-}
+HTTPD_PATH=${HTTPD_PATH:-}
+
+# Set up variables.
+TEST_UUID=$(uuidgen)
+IMAGE_KEY="osbuild-composer-ostree-test-${TEST_UUID}"
+
+# Set up temporary files.
+TEMPDIR=$(mktemp -d)
+BLUEPRINT_FILE=${TEMPDIR}/blueprint.toml
+SOURCE_FILE=${TEMPDIR}/copr_neptune.toml
+COMPOSE_START=${TEMPDIR}/compose-start-${IMAGE_KEY}.json
+COMPOSE_INFO=${TEMPDIR}/compose-info-${IMAGE_KEY}.json
+
+# Save some VARs for the next step
+cat >> /tmp/.env <<EOF
+IMAGE_KEY=${IMAGE_KEY}
+TEMPDIR=${TEMPDIR}
+EOF
+
+# Get the compose log.
+get_compose_log () {
+    COMPOSE_ID=$1
+    LOG_FILE=osbuild-${ID}-${VERSION_ID}-${COMPOSE_ID}.log
+
+    # Download the logs.
+    composer-cli compose log "$COMPOSE_ID" | tee "$LOG_FILE" > /dev/null
+}
+
+# Get the compose metadata.
+get_compose_metadata () {
+    COMPOSE_ID=$1
+    METADATA_FILE=osbuild-${ID}-${VERSION_ID}-${COMPOSE_ID}.json
+
+    # Download the metadata.
+    composer-cli compose metadata "$COMPOSE_ID" > /dev/null
+
+    # Find the tarball and extract it.
+    TARBALL=$(basename "$(find . -maxdepth 1 -type f -name "*-metadata.tar")")
+    tar -xf "$TARBALL" -C "${TEMPDIR}"
+    rm -f "$TARBALL"
+
+    # Move the JSON file into place.
+    cat "${TEMPDIR}"/"${COMPOSE_ID}".json | jq -M '.' | tee "$METADATA_FILE" > /dev/null
+}
+
+# Build ostree image.
+build_image() {
+    blueprint_file=$1
+    blueprint_name=$2
+
+    # Prepare the blueprint for the compose.
+    greenprint "📋 Preparing blueprint"
+    composer-cli blueprints push "$blueprint_file"
+    composer-cli blueprints depsolve "$blueprint_name"
+
+    # Start the compose.
+    greenprint "🚀 Starting compose"
+    composer-cli --json compose start "$blueprint_name" "$IMAGE_TYPE" | tee "$COMPOSE_START"
+    COMPOSE_ID=$(jq -r '.build_id' "$COMPOSE_START")
+
+    # Wait for the compose to finish.
+    greenprint "⏱ Waiting for compose to finish: ${COMPOSE_ID}"
+    while true; do
+        composer-cli --json compose info "${COMPOSE_ID}" | tee "$COMPOSE_INFO" > /dev/null
+        COMPOSE_STATUS=$(jq -r '.queue_status' "$COMPOSE_INFO")
+
+        # Is the compose finished?
+        if [[ $COMPOSE_STATUS != RUNNING ]] && [[ $COMPOSE_STATUS != WAITING ]]; then
+            echo ; greenprint "🚀 Finished compose"
+            break
+        fi
+        echo -n "."
+
+        # Wait 30 seconds and try again.
+        sleep 30
```

Comment:
Personally, I don't like using sleep. This is not a blocker for this PR. Just a general statement.

Reply:
This is how the osbuilder team and the qe team do it in their scripts. We need to wait for a while for the osbuild to finish the image. This is the easiest and more common way to do it.
Do you have a better alternative?

Reply:
A suggestion: mage Builder team to provide an API to return if the image is ready or not. If not, return in one of the header the expected amount of time to wait till the image is ready. The idea is inspired by the following (See the Retry-After header) https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429